### ISSUE TITLE
Create a list of comps

### ISSUE THREAD

**Issue by Krakaw** [big-neon/bn-api#422]
Child of big-neon/docs#19
Child of #421

Comps are also taken out of inventory but they have a flag set / their price set to 0 and they are transferred to their destination wallet

**Comment by Timo614** [comment]
Just confirming the behavior here / confirming the direction I've been going is correct for this one:

- An event is created with x number of tickets of varying ticket types
- An organization owner or member can create comps for a given ticket type which keeps the tickets in the organization's wallet and marks them as being Comp tickets which makes them unsellable
- The organization can transfer those tickets to another user

So they never go through the order process just are moved around as free tickets from the organization's wallet to the user in questions?

"Comps are also taken out of inventory" so from that I was thinking that ticket types have assets created for them and then the comps request marks a portion of them as unsellable and available for giving out as comps? Or should the logic work by allowing a comp request to create new assets as if they weren't included to begin with? Guess I'm wondering what the flow is here for the generation of these if we're taking them out of the available set or if the available set never included them prior to their addition if that makes sense.

**Comment by Timo614** [comment]
@zmelamed can you confirm the flow above?

**Comment by mikethetike** [comment]
I think it's important that the Order is still created for these comps, so that the financial reports can reflect them easily. The order's amount should just be 0. The API/front end shouldn't explicitly create an order, it should be created in the backend

**Comment by zmelamed** [comment]
Hey @Krakaw @Timo614 @mikethetike , I put together some flows here (also added this to big-neon/docs#19) to get across how this might work with turning a hold into a named list ... is this possible with our implementation... it's essentially taking a hold, and then adding a list of names which are assigned allotments of that hold (kind of holding part of the hold)... These names would be returned to the scanner app and box office to allow those tickets to be claimed or checked in without them having to be on the users device... Then eventually we could create a link to share with an external list admin (such as the band who has tickets held) where they could self manage the names on that "list" whcih is also described in this flow to some degree. 
![holds-lists-flows](https://user-images.githubusercontent.com/456953/47399648-56b40900-d6ee-11e8-9352-38a4a30f2578.png)

**Comment by mikethetike** [comment]
So a comp list is basically just a hold with a 100% discount?

**Comment by zmelamed** [comment]
Yes... a hold on tickets with a 100% discount and no fee. A held allotment of complimentary tix=a comp hold, and if you have a list of names attached to that you get a comp list.

**Comment by Timo614** [comment]
All this sounds good have been looking over the holds today and working through beginning a comp list aspect of it.

Based on the above description there's never going to be a case where a user has access to the comp ticket to then trade with others, etc. so it's a much simpler setup than I was originally thinking. Makes sense why it's setup this way from a box office perspective though.

> I think it's important that the Order is still created for these comps, so that the financial reports can reflect them easily. 

The comp list can be modified so I guess we should make the order at the time of comp use? So if only two of the four held comps for Keith in that mockup were used the free order would only reflect those two? Or were you thinking have the order reflect all those held aside regardless if they're ever used (and update that order as the comp list item is modified)?

For the purposes of the scope of this ticket:

1. Adding the ability for comp types of holds to be used (and making the discount in cents optional in that case) as they never go through checkout so it's not required
2. Treating comp holds differently than regular holds in that they can't be purchased just redeemed
3. Adding CRUD actions for the list manager to add new comps

Out of scope / another issue:

1. Ability to mark comp tickets as used / checkin via scanner app
2. Share comp list functionality

**Comment by Timo614** [comment]
Sounds good.

**Comment by Timo614** [comment]
I've started work on this here: https://github.com/Timo614/bn-api/tree/comps

To keep the scope constrained (and avoid too massive of a PR as the branch already getting pretty large and I haven't added tests yet) going to have the logic here primarily deal with the management of the comp lists themselves.

Outside of scope for this issue (will make followup issues after I push up the PR):

1. Redeeming a comp and triggering an order for comps still in the organization's possession (the first scenario Zach mentioned)
2. Transferring the comp to another user (second scenario Zach mentioned)
3. Adjusting purchasing to allow for zero fee, zero price cost for comp tickets via the redemption code flow (third scenario Zach mentioned, currently waiting on the hold purchasing PR but should be pretty simple)
